feat(STADTPULS-368): New Sensor Modal

vogelino opened the pull request:
This PR adds a Sensor modal component for the addition, edition, and deletion of a sensor.

dnsos commented on src/components/EditAddSensorModal/index.tsx:
```diff
@@ -0,0 +1,380 @@
+            <FormTextarea
+              {...field}
+              label='Beschreibung'
+              placeholder='Beschreibe kurz der Sensor'
```
Typo: der -> den

dnsos commented on src/components/EditAddSensorModal/index.tsx:
```diff
@@ -0,0 +1,380 @@
+                    setIntegration(newValue as "http" | "ttn");
+                  }}
+                  label='Integration'
+                  placeholder='Wie möchtest du dein Sensor integrieren?'
```
Typo: dein -> deinen

dnsos commented on src/components/EditAddSensorModal/index.tsx:
```diff
@@ -0,0 +1,380 @@
+                  {...field}
+                  onChange={newValue => {
+                    field.onChange(newValue);
+                    setIntegration(newValue as "http" | "ttn");
```
Isn't it possible to set this to `as definitions["sensors"]["connection_type"]`? This way we don't have to manually update this once the `connection_type`s change

vogelino replied:
The problem is that the backend currently also accepts "other" but the frontend doesn't. I will omit "other".

vogelino replied:
I tried omit but it isn't possible on enums... :(

dnsos commented on src/components/FormListBox/FormSelect.stories.tsx:
```diff
@@ -0,0 +1,188 @@
+import Funnel from "../../../public/images/icons/16px/funnel.svg";
+
+export default {
+  title: "Forms/FormListBox",
```
Great that this one is so flexible, to handle icons, default selections etc. 👍

dnsos commented on src/components/FormListBox/index.tsx:
```diff
@@ -0,0 +1,159 @@
+          }}
+        >
+          {label && (
+            <Listbox.Label className='block mb-2 cursor-pointer transition hover:opacity-60'>
```
So this is how we use `headless-ui`? It gives us a completely unstyled element and we use Tailwind to style it?

vogelino replied:
Exactly, and it takes care of all the aria labels, keyboard navigation, etc. Very cool!

dnsos commented on src/components/SensorPageHeader/withData.tsx:
```diff
@@ -0,0 +1,204 @@
+              title='Fehler!'
+              message={
+                <>
+                  Es ist beim editieren ein Fehler aufgetreten:{" "}
```
Maybe better: "Beim Editieren ist ein Fehler aufgetreten"

dnsos commented on src/components/SensorPageHeader/withData.tsx:
```diff
@@ -0,0 +1,204 @@
+          }
+        >
+          <div className='prose max-w-none'>
+            <p>Du bist dabeit der Sensor „{mergedSensor.name}“ zu löschen.</p>
```
Typos, should be: Du bist dabei, den Sensor [...] zu löschen.

vogelino replied:
OMG all the typo... Sorry. Danke :D

dnsos reviewed (changes requested):
Great work @vogelino!  This really seemed like a complex one. I tested the interaction and it works very nicely.

I have a few comments and questions and one display problem:

<img width="443" alt="Screenshot 2021-09-23 at 11 15 10" src="https://user-images.githubusercontent.com/15640196/134484117-dd0a6e99-cc8e-4cf0-a3e6-982ed92e44de.png">

The tooltip turns slightly transparent and is outside of the viewport. The latter is of course a problem with the tooltip itself, not this PR.

vogelino commented:
@dnsos I fixed the opacity of the tooltip (not the position, I would do that in another branch) and added the focus on the name input on mount. That should do the trick. Let me know what you think?

dnsos reviewed (approved):
Good changes, thank you! 👍 

(I also think the tooltip positioning should be fixed in another task)

tsboter commented:
:tada: This PR is included in version 1.0.0 :tada:

The release is available on [GitHub release](https://github.com/technologiestiftung/stadtpuls-frontend/releases/tag/v1.0.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
